src/utility/server_drv.cpp

```diff
@@ -110,16 +110,17 @@ void ServerDrv::startClient(uint32_t ipAddress, uint16_t port, uint8_t sock, uin
     SpiDrv::spiSlaveDeselect();
 }
 
-void ServerDrv::startClient(const char* host, uint8_t host_len, uint32_t ipAddress, uint16_t port, uint8_t sock, uint8_t protMode)
+void ServerDrv::startClient(const char* host, uint8_t host_len, uint32_t ipAddress, uint16_t port, uint8_t sock, uint8_t protMode, uint16_t timeout)
 {
     WAIT_FOR_SLAVE_SELECT();
     // Send Command
-    SpiDrv::sendCmd(START_CLIENT_TCP_CMD, PARAM_NUMS_5);
+    SpiDrv::sendCmd(START_CLIENT_TCP_CMD, PARAM_NUMS_6);
     SpiDrv::sendParam((uint8_t*)host, host_len);
     SpiDrv::sendParam((uint8_t*)&ipAddress, sizeof(ipAddress));
     SpiDrv::sendParam(port);
     SpiDrv::sendParam(&sock, 1);
-    SpiDrv::sendParam(&protMode, 1, LAST_PARAM);
+    SpiDrv::sendParam(&protMode, 1);
+    SpiDrv::sendParam(timeout, LAST_PARAM);
 
     // pad to multiple of 4
     int commandSize = 17 + host_len;
```

Comment:
```suggestion
    int commandSize = 19 + host_len;
```

Reply:
I got it:
header 5 bytes
hostname n + 1 (terminating zero)
ipaddress 4 bytes
port 2 bytes
sock 1 byte
protmode 1 byte
timeout 2 bytes 
end mark 1 byte

17 bytes + jostlen

Reply:
it is 17 bytes + hostlen:
header 5 bytes
hostname n + 1 (terminating zero)
ipaddress 4 bytes
port 2 bytes
sock 1 byte
protmode 1 byte
timeout 2 bytes 
end 1 byte

Reply:
The initial count seems correct, but my suggestion is wrong. It should be 20.
START CMD TOT LEN -> 3
host -> 1 + host_len
ipaddr -> 1 + 4
port -> 1 + 2
sock -> 1 + 1
protMode -> 1 + 1
timeout -> 1 + 2
END -> 1